components/support/nimbus-fml/src/main.rs

```diff
@@ -138,76 +194,429 @@ fn file_path(name: &str, args: &ArgMatches, cwd: &Path) -> Result<PathBuf> {
     }
 }
 
-pub struct GenerateStructCmd {
-    manifest: PathBuf,
-    output: PathBuf,
-    language: TargetLanguage,
-    load_from_ir: bool,
-    channel: String,
+pub(crate) enum CliCmd {
+    Generate(GenerateStructCmd),
+    DeprecatedGenerate(GenerateStructCmd, AboutBlock),
+    GenerateExperimenter(GenerateExperimenterManifestCmd),
+    GenerateIR(GenerateIRCmd),
 }
 
-pub struct GenerateExperimenterManifestCmd {
-    manifest: PathBuf,
-    output: PathBuf,
-    load_from_ir: bool,
-    channel: String,
-}
+#[cfg(test)]
+mod cli_tests {
+    use std::env;
 
-pub struct GenerateIRCmd {
-    manifest: PathBuf,
-    output: PathBuf,
-    load_from_ir: bool,
-    channel: String,
-}
+    use super::*;
 
-#[derive(Copy, Clone, Eq, PartialEq, Hash)]
-pub enum TargetLanguage {
-    Kotlin,
-    Swift,
-    IR,
-    ExperimenterYAML,
-    ExperimenterJSON,
-}
+    const FML_BIN: &str = "nimbus-fml";
+    const TEST_FILE: &str = "fixtures/fe/importing/simple/app.yaml";
+    const CACHE_DIR: &str = "./build/cache";
+    const REPO_FILE_1: &str = "./repos.versions.json";
+    const REPO_FILE_2: &str = "./repos.local.json";
+
+    fn package_dir() -> Result<PathBuf> {
+        let string = env::var("CARGO_MANIFEST_DIR")?;
+        Ok(PathBuf::try_from(string)?)
+    }
+
+    // All these tests just exercise the command line parsing.
+    // Each of the tests construct a command struct from the command line, and then
+    // test the command struct against the expected values.
 
-impl TargetLanguage {
-    #[allow(dead_code)]
-    pub(crate) fn extension(&self) -> &str {
-        match self {
-            TargetLanguage::Kotlin => "kt",
-            TargetLanguage::Swift => "swift",
-            TargetLanguage::IR => "fml.json",
-            TargetLanguage::ExperimenterJSON => "json",
-            TargetLanguage::ExperimenterYAML => "yaml",
+    ///////////////////////////////////////////////////////////////////////////
+    // In EXP-2560 we changed the command line to support importing and including.
+    // We still want to support the old command line, until we have time to update all the apps
+    // and the documentation, so we test that we can call them in the same way.
+    // The commands in these tests are taken from the apps (`nimbus-fml.sh` in FxiOS and
+    // `NimbusGradlePlugin.groovy` in AC).
+    #[test]
+    fn test_cli_legacy_generate_android_features() -> Result<()> {
```

Comment:
I absolutely love those backward compatible tests!! This gave me confidence that the refactoring didn't break the consumers!